File: python_solutions/950.py
from queue import SimpleQueue
class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        ind = SimpleQueue()
        for idx in range(len(deck)):
            ind.put(idx)
        deck.sort()
        new_deck = [-1 for i in range(len(deck))]
        for num in deck:
            idx = ind.get()
            new_deck[idx] = num
            if not ind.empty() :
                idx = ind.get()
                ind.put(idx)
        return new_deck

File: java_solutions/950.java
import java.util.* ; 
class Solution {
    public int[] deckRevealedIncreasing(int[] deck) {
        Queue<Integer> ind = new ArrayDeque<Integer>();
        Arrays.sort(deck) ; 
        for(int i = 0 ; i < deck.length ; i++){
            ind.offer(i);
        }
        int [] new_deck = new int[deck.length];
        for(var num : deck){
            var idx = ind.poll();
            new_deck[idx] = num ; 
            if (ind.size() > 0){
                idx = ind.poll();
                ind.offer(idx);
            }
        }
        return new_deck ; 
    }
}
